ChainRouter.php

```diff
@@ -255,26 +255,26 @@ public function generate($name, $parameters = array(), $absolute = UrlGeneratorI
      *
      * If the request context is not set, this simply returns the request object built from $uri.
      *
-     * @param string $uri
+     * @param string $pathinfo
      *
      * @return Request
      */
-    private function rebuildRequest($uri)
+    private function rebuildRequest($pathinfo)
     {
         if (!$this->context) {
-            return Request::create($uri);
+            return Request::create($pathinfo);
         }
 
         $server = array();
-        if ($this->context->getHost()) {
-            $server['SERVER_NAME'] = $this->context->getHost();
-            $server['HTTP_HOST'] = $this->context->getHost();
-        }
         if ($this->context->getBaseUrl()) {
-            $uri = $this->context->getBaseUrl().$uri;
+            $pathinfo = $this->context->getBaseUrl().$pathinfo;
             $server['SCRIPT_FILENAME'] = $this->context->getBaseUrl();
             $server['PHP_SELF'] = $this->context->getBaseUrl();
```

Comment:
the base url server variables are probably the only ones that you need to deal with as done here because that cannot be deducated from the URI alone.